src/partisan_amqp_peer_service_manager.erl

```diff
@@ -0,0 +1,411 @@
+update_members(_Nodes) ->
+    {error, not_implemented}.
+
+%% @doc Send message to a remote manager.
+send_message(Name, Message) ->
+    gen_server:call(?MODULE, {send_message, Name, Message}, infinity).
+
+%% @doc Cast a message to a remote gen_server.
+cast_message(Name, ServerRef, Message) ->
+    cast_message(Name, ?DEFAULT_CHANNEL, ServerRef, Message).
+
+%% @doc Cast a message to a remote gen_server.
+cast_message(Name, Channel, ServerRef, Message) ->
+    FullMessage = {'$gen_cast', Message},
+    forward_message(Name, Channel, ServerRef, FullMessage),
+    ok.
+
+%% @doc Forward message to registered process on the remote side.
+forward_message(Name, ServerRef, Message) ->
+    forward_message(Name, ?DEFAULT_CHANNEL, ServerRef, Message).
+
+%% @doc Forward message to registered process on the remote side.
+forward_message(Name, _Channel, ServerRef, Message) ->
+    gen_server:call(?MODULE, {forward_message, Name, ServerRef, Message}, infinity).
+
+%% @doc Receive message from a remote manager.
+receive_message(Message) ->
+    gen_server:call(?MODULE, {receive_message, Message}, infinity).
+
+%% @doc Attempt to join a remote node.
+join(Node) ->
+    gen_server:call(?MODULE, {join, Node}, infinity).
+
+%% @doc Attempt to join a remote node.
+sync_join(_Node) ->
+    {error, not_implemented}.
+
+%% @doc Leave the cluster.
+leave() ->
+    gen_server:call(?MODULE, {leave, node()}, infinity).
+
+%% @doc Remove another node from the cluster.
+leave(Node) ->
+    gen_server:call(?MODULE, {leave, Node}, infinity).
+
+%% @doc Decode state.
+decode(State) ->
+    sets:to_list(State).
+
+%% @doc Reserve a slot for the particular tag.
+reserve(_Tag) ->
+    {error, not_implemented}.
+
+%% @doc Inject a partition.
+inject_partition(_Origin, _TTL) ->
+    {error, not_implemented}.
+
+%% @doc Resolve a partition.
+resolve_partition(_Reference) ->
+    {error, not_implemented}.
+
+%% @doc Return partitions.
+partitions() ->
+    {error, not_implemented}.
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+
+%% @private
+-spec init([]) -> {ok, state_t()}.
+init([]) ->
+    %% Seed the process at initialization.
+    rand:seed(exsplus, {erlang:phash2([node()]),
+                        erlang:monotonic_time(),
+                        erlang:unique_integer()}),
+
+    %% Process connection exits.
+    process_flag(trap_exit, true),
+
+    %% Ensure that the amqp backend is started.
+    {ok, _} = application:ensure_all_started(amqp_client),
+
+    %% Initialize membership.
+    Myself = myself(),
+    {ok, Membership} = ?SET:mutate({add, Myself}, Myself, ?SET:new()),
+
+    %% Schedule periodic broadcast.
+    schedule_broadcast(),
+
+    URI = case os:getenv("RABBIT_URI") of
+        false ->
+            {stop, no_uri};
+        V ->
+            V
+    end,
+    
+    %% Open a connection to the local broker.
```

Comment:
I appreciate this is a prototype. :) What do you mean by "local" broker? I'm wondering what RabbitMQ topology is assumed. Would every partisan node connect to a centralised rabbit broker or use a local one somehow?

Reply:
That comment is wrong.

Reply:
Right now, I'm assuming that each one talks to a centralized broker of some sort for now.